pkg/controllers/core/catalogsource_controller.go

```diff
@@ -386,14 +388,45 @@ func (r *CatalogSourceReconciler) unpackJob(cs *corev1beta1.CatalogSource) *batc
 				},
 				Spec: v1.PodSpec{
 					RestartPolicy: v1.RestartPolicyOnFailure,
-					Containers: []v1.Container{
+					InitContainers: []v1.Container{
 						{
 							Image: r.OpmImage,
+							Name:  "initializer",
+							Command: []string{
+								"cp",
+								"/bin/opm",
+								fmt.Sprintf("%s%s", mountPath, "opm"),
+							},
+							VolumeMounts: []v1.VolumeMount{
+								{
+									Name:      opmVol,
+									MountPath: mountPath,
+								},
+							},
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Image: cs.Spec.Image,
 							Name:  "unpacker",
 							Command: []string{
-								"opm",
+								fmt.Sprintf("%s%s", mountPath, "opm"),
 								"render",
-								cs.Spec.Image,
+								"configs/",
```

Comment:
Probably need to do this or set working directory to `/`.
```suggestion
								"/configs/",
```

Reply:
Fair point, one thing I'm wondering is if we want to say that the `configs/` directory should always be in the root directory or just that the `configs/` directory exists within the default working directory? I'm fine with either but the latter gives users a bit more flexibility in how they structure the contents of their image versus the former of it always having to be in an exact location.

And maybe this is all moot since we have plans to implement a `spec.source` interface in the very near future where we could set this as a configurable field.

Thoughts?

Reply:
OLMv0 has flexibility for flexibility sake, and (at least from what I could tell) there was never a good reason explained for _why_ that flexibility (and therefore complexity) needed to exist.

So my stance is that we should assume that we'll find the FBC at `/configs`. Rukpak essentially splits this problem in two:
1. The source API doesn't know or care about the content of a filesystem, so it necessarily has knobs to essentially let you pick some subdirectory that should be extracted. The extracted directory eventually shows up in the `Bundle` provisioner as an `fs.FS` where the fs root is the contents of the extracted directory.
2. Anything interpreting the source, sees the fs. So it never even knows that this thing was a subdirectory of something else. It always looks like the root.

The result is that sources can have whatever structure they want, and provisioners can always just assume everything will be where they expect relative to the root of the fs.

Reply:
Right now, catalogd doesn't have this split concept, so I think we should make it as strict as we reasonably can. And once we introduce this split, then the source API can allow users to select a different subdirectory to find `configs`.

Reply:
https://github.com/operator-framework/catalogd/pull/54/commits/6108501183f8dde9108a13ae97af8c05bc72ec93